toolchain/sem_ir/formatter.cpp

```diff
@@ -212,14 +201,11 @@ class FormatterImpl {
 
   auto FormatInterface(InterfaceId id) -> void {
     const Interface& interface_info = sem_ir_.interfaces().Get(id);
+    FormatGeneric(interface_info.generic_id);
```

Comment:
Just a thought, but I've been mulling commonality in this code... if FormatInterfaceName etc were renamed to FormatName (or FormatIdAsName?) and relied on overload resolution, maybe a chunk of these functions could be moved into a template for code sharing?

(not for this PR, just wondering about your thoughts -- I've also been considering doing things like translating FormatInst to a switch instead of overloads)

Reply:
Turns out that this is directly useful for the new approach to generic formatting, so I went ahead and did it in this PR.